src/fluka/fluka_funcs.cpp

```diff
@@ -0,0 +1,1104 @@
+    return false;
+}
+
+
+/* testable wrapper function for f_look */
+int look(double& posx, double& posy, double& posz, double* dir, int& oldRegion) {
+  int flagErr;
+  int lattice_dummy;  // not used
+  int nextRegion;
+  f_look(posx, posy, posz, dir, oldRegion, lattice_dummy, nextRegion, flagErr, lattice_dummy);
+  return nextRegion;
+}
+
+
+/* determine where a particle is, given position, direction etc */
+void f_look(double& pSx, double& pSy, double& pSz,
+            double* pV, const int& oldReg, const int& oldLttc,
+            int& nextRegion, int& flagErr, int& newLttc) {
+  if (debug) {
+    std::cout << "-------f_look------" << std::endl;
+    std::cout << " stack size is " << flkstk_.npflka << std::endl;
+    std::cout << "position is " << pSx << " " << pSy << " " << pSz << std::endl;
+  }
+
+  reset_state(state);
+  // get the stack count
+  //  state.stack_count = flkstk_.npflka;
+
+  const double xyz[] = {pSx, pSy, pSz};       // location of the particle (xyz)
+  const double dir[] = {pV[0], pV[1], pV[2]};
+
+  int is_inside = 0;
+  int num_vols = DAG->num_entities(3);  // number of volumes
+
+  for (int i = 1 ; i <= num_vols ; i++) { // loop over all volumes
+    moab::EntityHandle volume = DAG->entity_by_index(3, i); // get the volume by index
+    // No ray history  - doesnt matter, only called for new source particles
+    moab::ErrorCode rval = DAG->point_in_volume(volume, xyz, is_inside, dir);
+    // check for non error
+    if (moab::MB_SUCCESS != rval)
+      fludag_abort("f_look", "DAGMC failed in point_in_volume", rval);
+
+    // we think that we are inside ,
+    if (is_inside == 1) {   // we are inside the cell tested
+      // test point in vol again, but reverse the direction
+      double new_dir[3];
+      new_dir[0] = -1.*dir[0];
+      new_dir[1] = -1.*dir[1];
+      new_dir[2] = -1.*dir[2];
+
+      int second_test = 0 ;
+
+      moab::ErrorCode rval = DAG->point_in_volume(volume, xyz, second_test, new_dir);
+      // check for non error
+      if (moab::MB_SUCCESS != rval)
+        fludag_abort("f_look", "DAGMC failed in point_in_volume", rval);
+
+      if (is_inside != second_test) {
+        if (debug) {
+          std::cout << "First test inconclusive, doing the slow test" << std::endl;
+        }
+
+        rval =  DAG->point_in_volume_slow(volume, xyz, second_test);
+        if (moab::MB_SUCCESS != rval)
+          fludag_abort("f_look", "DAGMC failed in point_in_volume_slow", rval);
+        // firing rays in two opposing directions didnt work, the slow test didnt work
+        if (second_test == 0) {
+          continue;
+        }
+      }
+
+      //WHEN WE ARE INSIDE A VOLUME, BOTH, nextRegion has to equal flagErr
+      nextRegion = i;
+      flagErr = nextRegion;
+
+      if (debug) {
+        std::cout << "region is " << nextRegion << " aka " << volume << std::endl;
+      }
+      return;
+    } else if (is_inside == -1) {
+      std::cout << "We cannot be here" << std::endl;
+      exit(0);
```

Comment:
If this is an error, I would write this error to `std::cerr` and call `exit(1)`.  Typical POSIX executable exit codes are (0 = success, anything else = failure)